packages/functional/src/__tests__/pipe-test.ts

```diff
@@ -0,0 +1,325 @@
+import { Base58EncodedAddress } from '@solana/addresses';
+import {
+    BaseTransaction,
+    Blockhash,
+    createTransaction,
+    setTransactionFeePayer,
+    setTransactionLifetimeUsingBlockhash,
+} from '@solana/transactions';
+
+import { pipe } from '../pipe';
+
+const feePayer =
+    '2eTCZxWZkU5h3Mo162gLRTECzuJhPgC1McB9rCcoqNm2' as Base58EncodedAddress<'2eTCZxWZkU5h3Mo162gLRTECzuJhPgC1McB9rCcoqNm2'>;
+const secondFeePayer =
+    '6JkwLherbVYPVF5sXGHm7qd9Lpd6gzinU4P792FkgdfS' as Base58EncodedAddress<'6JkwLherbVYPVF5sXGHm7qd9Lpd6gzinU4P792FkgdfS'>;
+
+const blockhashLifetime = { blockhash: '111' as Blockhash, lastValidBlockHeight: 5n };
+const secondBlockhashLifetime = { blockhash: '222' as Blockhash, lastValidBlockHeight: 8n };
+
+// The generic below is needed to shut TypeScript up and infer the proper return type.
+function setFeePayer<T extends BaseTransaction>(tx: T) {
+    return setTransactionFeePayer(feePayer, tx);
+}
```

Comment:
Will consumers have the same issue of having to wrap our functions like `setTransactionFeePayer` in their own function with this generic (or use arrow functions) to use them with `pipe`?

Reply:
Sort of, yeah. It depends on how dynamic they have to make their helper. For example, if they know they'll always be adding a fee payer to a `TransactionV0`, then no. But if it may change and/or may have other configs like a lifetime, then yeah kinda.

There's always `any`!